Internal: Update GitHub Community Documentation and Forms [ED-12633]

nicholaszein opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* **Updated** the Repository's **Readme** file `README.md` with revised text and replaced links with short links
* **Updated** the organization's **Contribution Guidelines** file `.github/CONTRIBUTING.md` with revised text and replaced links with short links
* **Created** the organization's **Code of Conduct** `.github/CODE_OF_CONDUCT.md` file
* **Deleted** the **deprecated** Bug Report issue template file `.github/ISSUE_TEMPLATE.md ` and the deprecated **Feature Request** issue template file `.github/feature-request.md`
* **Updated** the "New Issue's **Choose**" file `.github/ISSUE_TEMPLATE/config.yml` with revised text and replaced links with short links
* **Updated** the **Bug Report Form** template file `.github/ISSUE_TEMPLATE/bug.yml` with revised text and fields for better clarity and to better enforce our requirements for bug reporting
* **Updated** the **Feature Request Form** template file `.github/DISCUSSION_TEMPLATE/feature-request.yml` with revised text and fields for better clarity and to better enforce our requirements for bug reporting
* **Created** a new **Developers Community** template file `.github/DISCUSSION_TEMPLATE/developers-community.yml` to give context and better inform contributors.

## Description
An explanation of what is done in this PR

This is an update to all community-related files as well as the main repository's **readme** file to reflect better our current necessities as well as to replace outdated text and invalid links and add relevant information that was so far missing from those files.

All forms were also revised to reflect our requirements better.


## Test instructions
This PR can be tested by following these steps:

* Visit all documents
* Check if they are appearing correctly
* Review the text
* Test the links

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

nicholaszein committed:
[Internal] Update Readme.md with revised content and short-links

The whole document was revised.

All links were replaced with short links (elem.to) for more control over possible changes in destinations.

Some parts of the text were revised or replaced.

Other parts were added to include missing information.

nicholaszein committed:
[Internal] Updated Contributing.md with replaced text and links

The entire document was revised.

Part of the text was maintained, but most of it was rephrased to add more information and clarification.

All links were replaced with short links to help making quick changes without needing to change the document itself in case of changes to the links' destinations.

nicholaszein committed:
[Internal] Deleted deprecated files

Since we now use custom templates forms for both Issues and Feature Requests, we no longer need these files.

nicholaszein committed:
[Internal] Created new Code_of_conduct.md file

Introducing the new Code of Conduct that should cover all collaborative spaces in the Elementor GitHub organization.

nicholaszein committed:
[Internal] Updated config.yml

All links were replaced with short links for easy changes should the necessity arises.

New links were added.

All the descriptions were revised.

nicholaszein committed:
[Internal] Updated the Bug Report form

The entire form was reformulated to add more clarity and enforce our requirements in a more easy to understand way.

nicholaszein committed:
[Internal] Updated the Feature Request form

The entire form was reformulated to add more clarity and better enforce our requirements.

nicholaszein committed:
[Internal] Created a new form for the Developers Community Discussion category

Added a new form with relevant information.

rami-elementor commented on README.md:
```diff
@@ -1,47 +1,103 @@
 There are many ways to contribute to Elementor:
 
-**Bug Report**: If you find a bug in the source code, you can help us by [submitting a bug report](https://github.com/elementor/elementor/issues/new?template=Bug_report.md). Even better, you can [submit a Pull Request](https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md#before-submitting-a-pr) with a fix.
+#### Contribution Guidelines
```
You are jumping from H2 to H4?

nicholaszein replied:
Oops, you're right. I missed that one! I'm going to fix this.

rami-elementor commented on .github/CONTRIBUTING.md:
```diff
@@ -1,79 +1,132 @@
-# <p><a href="https://elementor.com/?utm_source=github-repo&utm_medium=link&utm_campaign=contribution">![logo](https://user-images.githubusercontent.com/1778512/191041718-728d179e-07cb-4cb4-953a-6c294ee8c4db.png)</a></p>
 
-Before filing a bug report, a feature request or sending pull requests, be sure to read the contribution guidelines.
+# <a href="https://elementor.com/?utm_source=github-repo&utm_medium=link&utm_campaign=contribution">![logo](https://user-images.githubusercontent.com/1778512/191041718-728d179e-07cb-4cb4-953a-6c294ee8c4db.png)</a>
```
Why use `#` ? double H1 headings...

nicholaszein replied:
I copied the header from the original file. So it came with two H1s. I've fixed it and will send another commit.

Thanks for spotting this!

nicholaszein committed:
Fixed semantical issues

Fixed headings that were used in the wrong places.

nicholaszein committed:
Merge branch 'elementor:main' into github-community

nicholaszein commented:
@rami-elementor,

there, I fixed the semantical problems! Thanks again!

https://github.com/elementor/elementor/pull/24159/commits/9fb96619a356dad310128d9eb2ff423ddfeb3687

nicholaszein commented:
The last commit wasn't supposed to be there. I might have inadvertently clicked to sync branches while making the new commit.

StyleShit commented on README.md:
```diff
@@ -1,47 +1,103 @@
+<br>
 
-It's time for **[Elementor Website Builder](https://elementor.com/?utm_source=github-repo&utm_medium=link&utm_campaign=readme)**.
+<p><a href="https://elemn.to/gh-to-elementor"><img src="https://ps.w.org/elementor/assets/screenshot-1.gif" alt="Elementor Website Builder"></a></p>
```
maybe use the GIF with the new UI if you're already here?

nicholaszein committed:
Replaced the cover animation with one with the new interface, pointing to a relative path

This commit replaces the old cover animation gif image of the Drag & Drop builder in action with a new one, with the new Editor UI. It also uses a relative path for the image inside the .github/assets folder, allowing us to change the image in the future if needed by just replacing the file with another one with the same name.

nicholaszein committed:
Merge branch 'github-community' of https://github.com/nicholaszein/elementor into github-community

arielk committed:
Merge branch 'main' into github-community

arielk committed:
Merge branch 'main' into github-community
